src/nvim/option_defs.h

```diff
@@ -1047,20 +1044,44 @@ typedef enum {
   PV_MAXVAL = 0xffff,  // to avoid warnings for value out of range
 } idopt_T;
 
+// Option value type
+typedef enum {
+  kOptValTypeNil = 0,
+  kOptValTypeBoolean,
+  kOptValTypeNumber,
+  kOptValTypeString,
+  // This represents the amount of option value types and must be kept at the end of the enum
+  kOptValTypeCount
```

Comment:
very common pattern in the codebase, don't need a long explanation
```suggestion
  kOptValTypeCount  // Must be last.
```